transport/httptransport/http_transport.go

```diff
@@ -216,53 +234,114 @@ type transfer struct {
 
 	client *http.Client
 	dl     *logs.DealLogger
+
+	nChunks int
 }
 
 func (t *transfer) execute(ctx context.Context) error {
 	duuid := t.dealInfo.DealUuid
-	for {
-		// construct request
-		req, err := http.NewRequest("GET", t.tInfo.URL, nil)
-		if err != nil {
-			return fmt.Errorf("failed to create http req: %w", err)
+
+	// create downloaders. Each downloader must be initialised with the same byte range across restarts in order to resume previous downloads.
+	// if the output file contains some data in it already, do not create a downloader for it again
```

Comment:
> if the output file contains some data in it already, do not create a downloader for it again

I'm not sure I understand this comment.. are you saying that if some chunks have already downloaded, don't download them again?

Reply:
There are a few possible scenarios there: 
1. some chunks might have been fully downloaded, while others just partially. We need to do nothing for the former and finish downloading for the latter. 
2. all chunks have been downloaded. Only some chunks have been merged into the main file and their temp files have been deleted. We need to make sure to not to re-download the merged chunks again. 

Added clarification to the code.